src/stringobj.c

```diff
@@ -415,10 +415,14 @@ static Obj FuncREVNEG_STRING(Obj self, Obj val)
 Obj NEW_STRING ( Int len )
 {
   Obj res;
-  if (len < 0)
-       ErrorQuit(
-           "NEW_STRING: Cannot create string of negative length %d",
-           (Int)len, 0L);
+  if (len < 0) {
+      ErrorQuit("NEW_STRING: Cannot create string of negative length %d",
+                (Int)len, 0L);
+  }
+  if (len > INT_INTOBJ_MAX) {
+      ErrorQuit("NEW_STRING: Size must be a small integer", 0L, 0L);
```

Comment:
```suggestion
      ErrorQuit("NEW_STRING: length must be a small integer", 0L, 0L);
```
We say "length" everywhere else, don't we?